animated-transformer/src/lib/transformer/transformer_gtensor.ts

```diff
@@ -358,11 +358,8 @@ export function computeAttnHead(
       .scalarDiv(makeScalar(Math.sqrt(seqInput.dim.inputRep.size), 'float32'));
   }
 
-  const maskedAffinities = ComputeMaskedAffinities(rawAttention); // Uncomment this and the next lines to compute Masked Attention
-  const attention = maskedAffinities.softmax('queryPos'); // Uncomment this and the previous lines to compute Masked Attention
-  //const attention = rawAttention.softmax('queryPos'); // Uncomment this line to compute Unmasked self attention
-
-  // Dropout on the attention weights.
+  const maskedAffinities = computeMaskedAffinities(rawAttention);
```

Comment:
Can you add a TODO here to, eventually, pass in precomputed attention mask to the function, rather than recompute attention masks inference pass.